notes/zrq/20210212-01-speed-tests.txt

```diff
@@ -0,0 +1,881 @@
+
+        Best practices for caching in Spark SQL
+        https://towardsdatascience.com/best-practices-for-caching-in-spark-sql-b22fb0f02d34
+
+        RADOS (Reliable Autonomic Distributed Object Store)
+        https://searchstorage.techtarget.com/definition/RADOS-Reliable-Autonomic-Distributed-Object-Store
+
+        CephFS: a new generation storage platform for Australian High Energy Physics
+        https://indico.cern.ch/event/505613/contributions/2230911/attachments/1345227/2039428/Oral-v5-162.pdf
+
+        CephFS file layouts
+        https://docs.ceph.com/en/mimic/cephfs/file-layouts/
+
+        Detecting CPU steal time in guest virtual machines
+        https://opensource.com/article/20/1/cpu-steal-time
+
+    Results:
+
+        Notebook works with 100% of eDR3 and 500 trees.
+        Need to experiment with adding more trees.
+
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+# Login to Zeppelin ...
+#[user@desktop]
+
+    firefox --new-window "http://zeppelin.metagrid.xyz:8080/" &
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+
+    Import notebooks from GitHub, clear the output and run all the cells ...
+
+    Good astrometric solutions via ML Random Forrest classifier
+    https://raw.githubusercontent.com/wfau/aglais-notebooks/main/2FRPC4BFS/note.json
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+
+Stelio's test #1a
+20210211-ML-Notebook-Benchmarking.txt
+
+    Cinder Volumes for temp storage for Spark & Hadoop
+    500 trees
+
+    main select statement
+	>  Took 43 mins
+
+    RandomForestClassifier - 10% data 500 trees
+	> Took 17 mins
+
+Stelio's test #1b
+20210211-ML-Notebook-Benchmarking.txt
+
+    Cinder Volumes for temp storage for Spark & Hadoop
+    5000 trees
+
+    main select statement
+	> ???
+
+    RandomForestClassifier - 10% data 5000 trees
+	> ???
+
+	notebook took 3 hrs 23 min 28 sec ()
+
+Stelio's test #2
+20210211-ML-Notebook-Benchmarking.txt
+
+    Revert changes to Ansible scripts so that it matches what is currently deployed on zeppelin.aglais.uk
+
+    main select statement
+    > 28 min 10 sec.
+
+    RandomForestClassifier - (assume 10% data 500 trees, not stated)
+    > 15 min 28 sec.
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+
+Live deployment #1
+
+    quick_filter=' AND MOD(random_index, 10) = 0'
+    quick_plot_filter=' AND MOD(random_index, 25) = 0'
+
+    main select statement
+    Took 29 min 10 sec. Last updated by gaiauser at February 12 2021, 4:59:01 AM.
+
+    first plot
+    ....
+
+    good/bad select
+    ....
+
+    RandomForestClassifier - 10% data 500 trees
+    Took 12 min 4 sec. Last updated by gaiauser at February 12 2021, 5:11:23 AM.
+
+
+Live deployment #2
```

Comment:
Was this run soon after the first one? Wondering any possible caching might have an effect on the result

Reply:
Yes - caching is an issue. Timings for different runs varies quite a lot.
Nigel's notebook has a call to clear the cache, but it doesn't clear everything. 
 
    %spark.pyspark
    # clear any previously cached data in the context ....
    sqlContext.clearCache()